### PR TITLE
Prac 5 code review request

### PR DESCRIPTION
**Pull request by JocoBuen**
@j-manio please leave comments or feedback for improvement and what did I do wrong.

### CODE REVIEW

File: prac_05/state_names.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"""
+CP1404/CP5632 Practical
+State names in a dictionary
+File needs reformatting
+"""
+
+# TODO: Reformat this file so the dictionary code follows PEP 8 convention
```
Comment (j-manio) [inline prac_05/state_names.py:7]: Do think this line needs to be here?

Comment (j-manio) [reply prac_05/state_names.py:7]: # TODO: Reformat this file so the dictionary code follows PEP 8 convention

File: prac_05/state_names.py [hunk 2]
```diff
@@ -5,14 +5,21 @@
...
 while state_code != "":
     if state_code in CODE_TO_NAME:
-        print(state_code, "is", CODE_TO_NAME[state_code])
+        print(f"{state_code:<3} is {CODE_TO_NAME[state_code]}")
     else:
         print("Invalid short state")
-    state_code = input("Enter short state: ")
+    state_code = input("Enter short state: ").upper()
+
+try:
```
Comment (j-manio) [inline prac_05/state_names.py:20]: Line 20, maybe put a comment on why did you use an exception?

File: prac_05/word_occurrences.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+def main():
+    text_to_number = {}
+    text = input("Text: ")
+    texts = text.split(" ")
+    texts.sort()
+    count_the_occurrences_of_words(text_to_number, texts)
+    display_the_word_occurrences(text_to_number)
+
+
+def display_the_word_occurrences(text_to_number):
```
Comment (j-manio) [inline prac_05/word_occurrences.py:17]: Line 17: Something is missing here that will explain what will the function do.

File: prac_05/word_occurrences.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    max_length = max(len(text) for text in text_to_number.keys())
+    for text, number in text_to_number.items():
+        print(f"{text:{max_length}} : {number}")
+
+
+def count_the_occurrences_of_words(text_to_number, texts):
```
Comment (j-manio) [inline prac_05/word_occurrences.py:23]: Line 23: Explain how will the function will work.

File: prac_05/word_occurrences.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+Word Occurrences
+Estimate: 40 minutes
+Actual:   61 minutes
+"""
+
+
+def main():
+    text_to_number = {}
+    text = input("Text: ")
+    texts = text.split(" ")
```
Comment (j-manio) [inline prac_05/word_occurrences.py:11]: Probably use something that we always use when we do split(). Like parts?

File: prac_05/emails.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        email_to_name[email] = name
+        email = input("Email: ")
+
+    print()
+    for email, name in email_to_name.items():
+        print(f"{name} ({email})")
+
+
+def convert_email_to_name(email):
+    parts = email.split("@")
```
Comment (j-manio) [inline prac_05/emails.py:26]: Line 25: Explain what will the function will do.

File: prac_05/emails.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+
+def main():
+    email_to_name = {}
+
+    email = input("Email: ")
+    while email != "":
+        name = convert_email_to_name(email)
+        action = input(f"Is your name {email_to_name.get(email)}? (Y/N) ").upper()
+        if action == "N" or action == "NO":
```
Comment (j-manio) [inline prac_05/emails.py:15]: Could work but, have you read the wording in the practical? 
"In our program, if the user does not press Enter or Y, then ask for their name"

File: prac_05/wimbledon.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    champion_to_count = count_the_champions(records)
+    countries = get_the_country_of_champions(records)
+    for champion, count in champion_to_count.items():
+        print(f"{champion} {count}")
+    print(f"\nThese {len(countries)} have won Wimbledon:")
+    for country in countries:
+        print(country, end=", ")
+
+
+def get_the_country_of_champions(records):
```
Comment (j-manio) [inline prac_05/wimbledon.py:20]: Line 20: Explain what will the function will do.

File: prac_05/wimbledon.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    countries = set()
+    for record in records:
+        countries.add(record[1])
+    return countries
+
+
+def count_the_champions(records):
```
Comment (j-manio) [inline prac_05/wimbledon.py:27]: Line 27: Explain what will the function will do.

File: prac_05/wimbledon.py [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    champion_to_count = {}
+    for record in records:
+        count = champion_to_count.get(record[2], 0)
+        champion_to_count[record[2]] = count + 1
+    return champion_to_count
+
+
+def get_records(filename):
```
Comment (j-manio) [inline prac_05/wimbledon.py:35]: Line 35: Explain what will the function will do.

File: prac_05/wimbledon.py [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    countries = set()
+    for record in records:
+        countries.add(record[1])
+    return countries
+
+
+def count_the_champions(records):
+    champion_to_count = {}
+    for record in records:
+        count = champion_to_count.get(record[2], 0)
```
Comment (j-manio) [inline prac_05/wimbledon.py:30]: Maybe put a comment on what will this line do?

File: prac_05/wimbledon.py [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    for champion, count in champion_to_count.items():
+        print(f"{champion} {count}")
+    print(f"\nThese {len(countries)} have won Wimbledon:")
+    for country in countries:
+        print(country, end=", ")
+
+
+def get_the_country_of_champions(records):
+    countries = set()
+    for record in records:
```
Comment (j-manio) [inline prac_05/wimbledon.py:22]: Maybe put a comment to explain or have a quick definition what this for loop for?

File: prac_05/word_occurrences.py (hunk 5 above)
Comment (JocoBuen) [reply prac_05/word_occurrences.py:11]: parts is not applicable for this event.

**Review by JocoBuen — COMMENTED** [review]
Resolved all the feedback

**Pull request merged**